tokio-postgres/src/connect.rs

```diff
@@ -32,11 +32,12 @@ where
             .copied()
             .unwrap_or(5432);
 
-        let hostname = match host {
-            Host::Tcp(host) => host.as_str(),
+        let hostname = match (config.tls_host.as_deref(), host) {
+            (Some(tls_host), Host::Tcp(_)) => tls_host,
+            (None, Host::Tcp(host)) => host.as_str(),
             // postgres doesn't support TLS over unix sockets, so the choice here doesn't matter
```

Comment:
I believe this is also the reason we shouldn't go down the unix domain route for pg

Reply:
Ah yeah good point.